insights/insightsclient/insightsclient.go

```diff
@@ -82,8 +81,23 @@ func New(client *http.Client, maxBytes int64, metricsName string, operatorName s
 		operatorName: operatorName,
 		mimeType:     mimeType,
 		authorizer:   authorizer,
-		clusterInfo:  clusterInfo,
+		kubeConfig:   kubeConfig,
+	}
+}
+
+// Get Cluster Version via API
+func (c *Client) getClusterVersion() (*configv1.ClusterVersion, error) {
+	if c.clusterVersion != nil {
+		return c.clusterVersion, nil
+	}
+	ctx := context.Background()
+	client, err := configv1client.NewForConfig(c.kubeConfig)
```

Comment:
I think it would be better to check the `err` here as well, because the `client` will be likely nil in case of some error and it will probably panic on the line below.

Reply:
Updated.